num_enum_derive/src/lib.rs

```diff
@@ -248,7 +248,20 @@ impl Parse for EnumInfo {
 
             let mut next_discriminant = literal(0);
             for variant in data.variants.into_iter() {
+                let mut span = variant.span();
+                // Sadly `Span::join` is a nightly-only API at the moment - see https://github.com/rust-lang/rust/issues/54725
+                if let Some(span_with_fields) = span.join(variant.fields.span()) {
+                    span = span_with_fields;
+                }
                 let ident = variant.ident;
+                let fields = variant.fields;
+
+                match fields {
+                    Fields::Named(_) | Fields::Unnamed(_) => {
```

Comment:
Tiny nit: since tuple variants don't really have _fields_ / or rather, it may not be seen as such by some users, I'd replace the `fields` mention for `Fields::Unnamed` with something else. Maybe something like associated data / payloads, or even simpler: "only unit variants are supported". Since this is subjective, feel free to disregard this 🙂

Reply:
Great feedback - thanks!